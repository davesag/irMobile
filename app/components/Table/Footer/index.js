import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { Text } from 'react-native-elements'

import styles from '../styles'

const makeKey = (text, index) => `${text}-${index}`

const Footer = ({ data }) =>
  data ? (
    <View style={styles.row}>
      {data.map((text, i) => (
        <View key={makeKey(text, i)} style={styles.footerCell}>
          <Text style={styles.footer}>{text}</Text>
        </View>
      ))}
    </View>
  ) : null

Footer.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string)
}

Footer.defaultProps = {
  data: null
}

export default Footer
