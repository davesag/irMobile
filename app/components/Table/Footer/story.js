import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { SafeAreaView, View } from 'react-native'

import Footer from '.'
import styles from '../styles'

const data = ['Total', '', '', '12,345.67']

storiesOf('Table.Footer', module)
  .add('without data', () => (
    <SafeAreaView>
      <View style={styles.table}>
        <Footer />
      </View>
    </SafeAreaView>
  ))
  .add('with data', () => (
    <SafeAreaView>
      <View style={styles.table}>
        <Footer data={data} />
      </View>
    </SafeAreaView>
  ))
