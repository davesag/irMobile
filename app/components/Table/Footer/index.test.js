import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Footer from '.'

let tree

describe('rendering', () => {
  describe('without data', () => {
    beforeAll(() => {
      tree = renderer.create(<Footer />)
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })

  describe('with data', () => {
    const data = ['Total', '', '12345.00']

    beforeAll(() => {
      tree = renderer.create(<Footer data={data} />)
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })
})
