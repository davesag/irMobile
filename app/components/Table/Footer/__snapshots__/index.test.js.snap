// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with data rendered correctly  1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "paddingTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "center",
        "borderBottomWidth": 0.5,
        "borderTopWidth": 0.5,
        "flexWrap": "nowrap",
        "height": 25,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
        "width": "25%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "center",
        "borderBottomWidth": 0.5,
        "borderTopWidth": 0.5,
        "flexWrap": "nowrap",
        "height": 25,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
        "width": "25%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "center",
        "borderBottomWidth": 0.5,
        "borderTopWidth": 0.5,
        "flexWrap": "nowrap",
        "height": 25,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
        "width": "25%",
      }
    }
  />
</View>
`;

exports[`rendering without data rendered correctly  1`] = `null`;
